examples/rust-simple/src/main.rs

```diff
@@ -42,57 +30,85 @@ fn main() {
     });
 }
 
-fn run() -> Result<(), String> {
+async fn run() -> Result<(), String> {
     println!("Creating SDK instance");
-    let sdk = agones::Sdk::new().map_err(|_| "Could not connect to the sidecar. Exiting!")?;
-
-    let _health = thread::spawn(enclose! {(sdk) move || {
-        loop {
-            match sdk.health() {
-                (s, Ok(_)) => {
-                    println!("Health ping sent");
-                    sdk = s;
+    let mut sdk = agones::Sdk::new(None /* default port */, None /* keep_alive */)
+        .await
+        .map_err(|e| format!("unable to create sdk client: {}", e))?;
+
+    let health = sdk.health_check();
```

Comment:
Maybe I'm misunderstanding something in this one - but shouldn't we be running 

```rust
    health
        .send(())
        .await
        .map_err(|_| "health receiver closed".to_owned())?;
```

On a regular interval, so that the example server remains healthy after being spun up?

Reply:
Yes, it should and _now_ I am remember why I felt it was ok to do that `spawn_health_task` thing I had before, as the original example spawned a separate thread for the health check, which actually had the problem that if the game had become unhealthy (other than the main thread shutting down and thus the entire application), that thread would have kept running and kept sending health checks, but the `spawn_health_task` method I had would have been dropped and thus cancelled the task sending the health checks.

Anyways, I'll replicate what the original example was doing, as that keep it closer to the old API, but will have the expected semantics that eg a panic on the thread will drop the task and stop sending the health check.